### ISSUE TITLE
Cannot spawn more than one thread

### ISSUE THREAD

**Issue by hatala91** [geckosio/phaser-on-nodejs#10]
**Description**
I am using `phaser-on-nodejs` for implementing an authoritative server instance of a multiplayer online game. There might be multiple games in parallel, which is why I need multiple instances of the server and I was hoping to solve this with multithreading.

**How to reproduce**
Assuing I have a working `phaser-on-nodejs` game working with an entry script called `game.js`, I spawn the server side code using something like:

```
import { Worker } from "worker_threads";
const gameProcess = new Worker(`game.js`, {
  workerData: gameParameters,
});
```

With the first thread this works fine, however on spawning a second thread I get:
```
Error: Module did not self-register: '/Users/sebastianhatala/dev/realtime-multiplayer-pong-backend/node_modules/canvas/build/Release/canvas.node'.
Process exited with exit code 1
```

**Comment by yandeu** [comment]
Looks like the error has something to do with [`node-canvas`](https://github.com/Automattic/node-canvas). Have you already check the node-canvas repo?

Maybe this helps: https://github.com/Automattic/node-canvas/issues/1718

**Comment by hatala91** [comment]
Thanks for looking into this, turns out that the causing issue is indeed `node-canvas` but another issue: https://github.com/Automattic/node-canvas/issues/1394

Closing this, though of course a canvas independent implementation would be amazing - maybe with Phaser 4.

**Comment by yandeu** [comment]
👍🏻

---

Btw, I was experimenting with a Standalone Arcade Physics lib.
Maybe it helps https://github.com/yandeu/arcade-physics_dev

**Comment by yandeu** [comment]
No, matter.js is not abandoned. It's my favorite 2d physics engine.

**Comment by hatala91** [comment]
Okay, glad to hear that too! I had read that somewhere (cannot have been a credible source though) and the open issues/ PRs did not directly contradict it. But we are way off topic here now anyway 😉
